File: 7Modules/Jacob/backend/schemas/inventory.py
from typing import List, Optional

try:
    from pydantic import BaseModel, ConfigDict

    class ORMBaseModel(BaseModel):
        model_config = ConfigDict(from_attributes=True)

except ImportError:  # pragma: no cover
    from pydantic import BaseModel

    class ORMBaseModel(BaseModel):
        class Config:
            orm_mode = True


class PartCategoryCreate(ORMBaseModel):
    name: str
    subcategory: Optional[str] = None


class PartCategoryResponse(ORMBaseModel):
    id: int
    name: str
    subcategory: Optional[str] = None


class CatalogEntryCreate(ORMBaseModel):
    category_id: int
    part_name: str
    part_number: str
    brand: Optional[str] = None
    unit_price: float
    quantity_in_stock: int
    min_order_quantity: int = 1
    lead_time_hours: int


class CatalogEntryUpdate(ORMBaseModel):
    category_id: Optional[int] = None
    part_name: Optional[str] = None
    part_number: Optional[str] = None
    brand: Optional[str] = None
    unit_price: Optional[float] = None
    quantity_in_stock: Optional[int] = None
    min_order_quantity: Optional[int] = None
    lead_time_hours: Optional[int] = None


class CatalogEntryResponse(ORMBaseModel):
    id: int
    supplier_id: int
    category_id: int
    part_name: str
    part_number: str
    normalized_part_number: str
    brand: Optional[str] = None
    unit_price: float
    quantity_in_stock: int
    min_order_quantity: int
    lead_time_hours: int
    created_at: Optional[str] = None
    updated_at: Optional[str] = None
    supplier_business_name: Optional[str] = None
    category_name: Optional[str] = None
    distance_km: Optional[float] = None


class InventoryTransactionResponse(ORMBaseModel):
    catalog_id: int
    change_amount: int
    reason: str
    created_at: Optional[str] = None


class CSVUploadError(ORMBaseModel):
    row_number: int
    error: str
    row_data: Optional[dict] = None


class CSVUploadResponse(ORMBaseModel):
    total_rows: int
    successful: int
    failed: int
    errors: List[CSVUploadError]


class CatalogListResponse(ORMBaseModel):
    items: List[CatalogEntryResponse]
    page: int
    page_size: int
    total: int

File: 7Modules/Jacob/frontend/src/types/inventory.ts
export type PartCategory = {
  id: number;
  name: string;
  subcategory?: string | null;
};

export type CatalogEntry = {
  id: number;
  supplier_id: number;
  category_id: number;
  part_name: string;
  part_number: string;
  normalized_part_number: string;
  brand?: string | null;
  unit_price: number;
  quantity_in_stock: number;
  min_order_quantity: number;
  lead_time_hours: number;
  created_at?: string | null;
  updated_at?: string | null;
  supplier_business_name?: string | null;
  category_name?: string | null;
  distance_km?: number | null;
};

export type CatalogListResponse = {
  items: CatalogEntry[];
  page: number;
  page_size: number;
  total: number;
};

export type InventoryTransaction = {
  catalog_id: number;
  change_amount: number;
  reason: string;
  created_at?: string | null;
};

export type CSVUploadError = {
  row_number: number;
  error: string;
  row_data?: Record<string, string> | null;
};

export type CSVUploadResponse = {
  total_rows: number;
  successful: number;
  failed: number;
  errors: CSVUploadError[];
};

export type SupplierProfile = {
  id: number;
  user_id: number;
  business_name: string;
  warehouse_address?: string | null;
  gst_number?: string | null;
  service_radius_km: number;
  latitude: number;
  longitude: number;
  reliability_score: number;
  distance_km?: number | null;
};

export type SupplierSummary = SupplierProfile & {
  total_parts: number;
  total_stock_value: number;
  avg_lead_time_hours: number;
};

export type CartItem = {
  catalog_id: number;
  part_name: string;
  part_number: string;
  supplier_name: string;
  unit_price: number;
  quantity: number;
};
